scripts/reporting/statsGenElastic/utils.js

```diff
@@ -0,0 +1,137 @@
+/**
+*	Copyright (C) 2019 3D Repo Ltd
+*
+*	This program is free software: you can redistribute it and/or modify
+*	it under the terms of the GNU Affero General Public License as
+*	published by the Free Software Foundation, either version 3 of the
+*	License, or (at your option) any later version.
+*
+*	This program is distributed in the hope that it will be useful,
+*	but WITHOUT ANY WARRANTY; without even the implied warranty of
+*	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*	GNU Affero General Public License for more details.
+*
+*	You should have received a copy of the GNU Affero General Public License
+*	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict'
+const dateFormat = require('dateformat');
+const fs = require('fs');
+const concat = require('concat-files');
+const Utils = {};
+const CryptoJS = require("crypto-js");
+
+Utils.hashCode = function(s){
+	return CryptoJS.MD5(s).toString();              
+  }
+
+Utils.concat = (files, dest) => {
+	return new Promise((resolve, error) => {
+		concat(files, dest, (err) => {
+			if(err) error(err);
+			else resolve();
+		});
+	});
+}
+
+Utils.formatDate = (date) => {
+	return date.toISOString()
+	// return dateFormat(date, 'dd-mm-yy'); // old format
+}
+
+Utils.generateFileName = (prefix) => {
+	const date = Utils.formatDate(new Date());
+	return `${prefix}_${date}.csv`;
+}
+
+Utils.mkdir = (dir) => {
+	return new Promise((resolve, error) => {
+		fs.mkdir(dir, {recursive: true}, (err) => {
```

Comment:
this is just `fs.promise.mkdir`?

Reply:
I didn't write this bit, don't need it at all in this version but came through from the file based one.